relay/http.go

```diff
@@ -188,6 +207,9 @@ func (h *HTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	for _, b := range h.backends {
 		b := b
+		if b.httpauthmode == "pass" {
+			b.authorization = r.Header.Get("Authorization")
```

Comment:
This should use [`req.BasicAuth`](https://golang.org/pkg/net/http/#Request.BasicAuth).